src/core/radio/radio.cpp

```diff
@@ -92,4 +95,128 @@ Error Radio::Transmit(Mac::TxFrame &aFrame)
 }
 #endif // OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
 
+#if OPENTHREAD_CONFIG_RADIO_STATS_ENABLE && (OPENTHREAD_FTD || OPENTHREAD_MTD)
+RadioStatistics::RadioStatistics(void)
+    : mStatus(kDisabled)
+    , mSleepTime(0)
+    , mRxTime(0)
+    , mTxTime(0)
+    , mLastCalTime(0)
+{
+}
+
+void RadioStatistics::ChangeStatus(Status aStatus)
+{
+    TimeMicro curTime = TimerMicro::GetNow();
+    if (mStatus != kDisabled)
+    {
+        UpdateTime(curTime);
+    }
+    else
+    {
+        mLastCalTime = TimerMicro::GetNow();
+    }
+
+    mStatus = aStatus;
+}
+
+void RadioStatistics::HandleReceiveAt(uint32_t aDurationUs)
+{
+    // The actual rx time of ReceiveAt cannot be obtained from software level. This is a workaround.
+    if (mStatus == kSleep)
+    {
+        mRxTime += aDurationUs;
+    }
+}
+
+void RadioStatistics::TxDone(otError aError, uint16_t aBytes)
+{
+    if (aError == kErrorNone || aError == kErrorNoAck)
+    {
+        uint32_t txTimeUs = (aBytes + Mac::Frame::kPhyHeaderSize) * Radio::kSymbolsPerOctet * Radio::kSymbolTime;
+        UpdateTime(TimerMicro::GetNow());
+        mTxTime += txTimeUs;
+        if (mStatus == kReceive)
+        {
+            if (mRxTime > txTimeUs)
+            {
+                mRxTime -= txTimeUs;
+            }
+            else
+            {
+                mRxTime = 0;
+            }
+        }
+        else if (mStatus == kSleep)
```

Comment:
Do we need to add the `Ack` RX time to the `mRxTime` when the `mStatus` is `kSleep`?